packages/orderbook/README.md

```diff
@@ -41,455 +41,463 @@ npm install @rainlanguage/orderbook
+const [firstDeployment] = deploymentsResult.value;
+
+const guiResult = await DotrainOrderGui.newWithDeployment(dotrainWithGui, firstDeployment);
+if (guiResult.error) throw new Error(guiResult.error.readableMsg);
 const gui = guiResult.value;
 
-// Gui is an instance of DotrainOrderGui
-// You can use the gui to fully control the order deployment process
+const configResult = gui.getAllGuiConfig();
+if (configResult.error) throw new Error(configResult.error.readableMsg);
+const config = configResult.value;
 
-// Get the tokens that need to be selected before deploying the order
 const selectTokensResult = gui.getSelectTokens();
-if (selectTokensResult.error) {
-  console.error("Failed to get select tokens:", selectTokensResult.error.readableMsg);
-  return;
-}
+if (selectTokensResult.error) throw new Error(selectTokensResult.error.readableMsg);
 const selectTokens = selectTokensResult.value;
-// Returns tokens that need addresses set:
-// [{ key: "input-token", label: "USDT" }, { key: "output-token", label: "ETH" }]
 
-// Set the token addresses for the deployment
-gui.setSelectToken("input-token", "0x96B41289D90444B8ADD57e6F265DB5aE8651DF29"); // USDT on Flare
-gui.setSelectToken("output-token", "0x1D80c49BbBCd1C0911346656B529DF9E5c2F783d"); // WFLR on Flare
-
-// Get all the configuration needed to construct the UI
-const configResult = gui.getAllGuiConfig();
-if (configResult.error) {
-  console.error("Failed to get all gui config:", configResult.error.readableMsg);
-  return;
+const depositsResult = gui.getDeposits();
+if (depositsResult.error) throw new Error(depositsResult.error.readableMsg);
+const deposits = depositsResult.value;
+
+await gui.setSelectToken('input-token', '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'); // USDC
+await gui.setSelectToken('output-token', '0x4200000000000000000000000000000000000006'); // WETH
+const fieldResult = gui.setFieldValue('fixed-io', '1850');
+if (fieldResult.error) throw new Error(fieldResult.error.readableMsg);
+const amountResult = gui.setFieldValue('amount-per-trade', '250');
+if (amountResult.error) throw new Error(amountResult.error.readableMsg);
+await gui.setDeposit('usdc', '5000');
+const vaultIdResult = gui.setVaultId('input', 'usdc', '42');
+if (vaultIdResult.error) throw new Error(vaultIdResult.error.readableMsg);
+
+const allowancesResult = await gui.checkAllowances('0xOwner');
+if (allowancesResult.error) throw new Error(allowancesResult.error.readableMsg);
+const allowances = allowancesResult.value;
+
+const approvalCalldatasResult = await gui.generateApprovalCalldatas('0xOwner');
+if (approvalCalldatasResult.error) throw new Error(approvalCalldatasResult.error.readableMsg);
+
+const depositCalldatasResult = await gui.generateDepositCalldatas();
+if (depositCalldatasResult.error) throw new Error(depositCalldatasResult.error.readableMsg);
+
+const deploymentArgsResult = await gui.getDeploymentTransactionArgs('0xOwner');
```

Comment:
it would be good to explicitly say what to do with the deployment args, e.g. show an example where they get signed by a wallet.

also, we should show how to get the newly deployed order using `getAddOrdersForTransaction` on the client